tests/auto_shift/test_auto_shift.cpp

```diff
@@ -68,3 +68,30 @@ TEST_F(AutoShift, key_release_after_timeout) {
     run_one_scan_loop();
     VERIFY_AND_CLEAR(driver);
 }
+
+// test auto shift and space cadet interaction
+// press shift, press key, release shift, release key
+// the right interaction is we only get the shifted key
+// the wrong interaction is we get a bracket and a shifted key
+TEST_F(AutoShift, auto_shift_with_space_cadet) {
+    TestDriver driver;
+    InSequence s;
+    auto       left_shift = KeymapKey(0, 0, 0, SC_LSPO);
+    auto       key_a      = KeymapKey(0, 1, 0, KC_A);
+
+    set_keymap({left_shift, key_a});
+
+    /* Press regular key */
+    EXPECT_NO_REPORT(driver);
+    left_shift.press();
+    key_a.press();
+    left_shift.release();
+    key_a.release();
+    VERIFY_AND_CLEAR(driver);
+
+    /* Release regular key */
+    EXPECT_REPORT(driver, (KC_LSFT, KC_A));
```

Comment:
I would recommend adding the expected report calls during the keypresses, so it checks as they update. 

The current flow will error out, because by the time it's checked, there is no report.
```
[ RUN      ] AutoShift.auto_shift_with_space_cadet
tests/auto_shift/test_auto_shift.cpp:93: Failure
Actual function call count doesn't match EXPECT_CALL((driver), send_keyboard_mock(KeyboardReport (KC_LSFT, KC_A)))...
         Expected: to be called once
           Actual: never called - unsatisfied and active
tests/auto_shift/test_auto_shift.cpp:94: Failure
Actual function call count doesn't match EXPECT_CALL((driver), send_keyboard_mock(KeyboardReport ()))...
         Expected: to be called once
           Actual: never called - unsatisfied and active
AutoShift.auto_shift_with_space_cadet failed!
[ LEVEL    ] [TIME] [EVENT]
[ INFO     ] 0      tapping term is 200ms
[ TRACE    ] 0      pressed:  QK_SPACE_CADET_LEFT_SHIFT_PARENTHESIS_OPEN
[ TRACE    ] 0      pressed:  KC_A
[ TRACE    ] 0      released: QK_SPACE_CADET_LEFT_SHIFT_PARENTHESIS_OPEN was pressed for 0ms
[ TRACE    ] 0      released: KC_A was pressed for 0ms
[ TRACE    ] 0      1 keyboard task loop
[ INFO     ] 1      test fixture clean-up start.
[ TRACE    ] 1      2000 keyboard task loops
[ TRACE    ] 2001   2000 keyboard task loops
[ INFO     ] 4001   test fixture clean-up end.
[  FAILED  ] AutoShift.auto_shift_with_space_cadet (0 ms)
```

Reply:
Thanks for all the help here!

Sorry, I haven't really had time recently to finish fixing these tests. I will get to it at some point